addons/website_sale/tests/test_sale_process.py

```diff
@@ -331,3 +337,36 @@ def test_05_create_so_with_website_and_multi_company(self):
         # Different company on the SO and the env user company
         with self.assertRaises(ValueError, msg="Should not be able to create SO with company different than the website company"):
             self._create_so(self.demo_partner.id, self.company_c.id)
+
+    def test_06_new_user_address_state(self):
```

Comment:
For this test, you will also need to check the value of the so's partner, check the values of the billing/shipping addresses before creating a new one, then use a different address, and last but not least check the addresses are indeed different/the new one is correctly set up. =]